fedimint-server/src/config/api.rs

```diff
@@ -447,29 +452,23 @@ impl ConfigGenState {
 
         Ok(ConfigGenParams { local, consensus })
     }
+}
 
-    fn get_verification(&self) -> ApiResult<BTreeMap<PeerId, sha256::Hash>> {
-        let config = self
-            .config
-            .clone()
-            .ok_or(ApiError::bad_request("Missing config".to_string()))?;
-
-        let mut hashes = BTreeMap::new();
-        for (peer, cert) in config.consensus.tls_certs.iter() {
-            let mut engine = HashEngine::default();
-            let client_config_resp = config
-                .consensus
-                .try_to_config_response(&self.settings.registry)
-                .expect("hashes");
+pub fn get_verification_hashes(config: &ServerConfig) -> BTreeMap<PeerId, sha256::Hash> {
+    let mut hashes = BTreeMap::new();
+    for (peer, cert) in config.consensus.tls_certs.iter() {
+        let mut engine = HashEngine::default();
 
-            client_config_resp.client_config.consensus_hash();
-            cert.consensus_encode(&mut engine).expect("hashes");
+        config
+            .consensus
+            .consensus_encode(&mut engine)
+            .expect("hashes");
+        cert.consensus_encode(&mut engine).expect("hashes");
 
-            let hash = sha256::Hash::from_engine(engine);
-            hashes.insert(*peer, hash);
-        }
-        Ok(hashes)
+        let hash = sha256::Hash::from_engine(engine);
```

Comment:
Side-note on efficient hashing: by cloning the hash engine after feeding the config into it but before encoding the particular peer's cert, we could save on hashing and encoding (not encode the config in the loop but outside).